src/TableFunctions/TableFunctionObjectStorage.h

```diff
@@ -0,0 +1,166 @@
+                                      " - storage_account_url, container_name, blobpath, account_name, account_key, format, compression\n"
+                                      " - storage_account_url, container_name, blobpath, account_name, account_key, format, compression, structure\n";
+};
+
+struct S3Definition
+{
+    static constexpr auto name = "s3";
+    static constexpr auto storage_type_name = "S3";
+    static constexpr auto signature = " - url\n"
+                                      " - url, format\n"
+                                      " - url, format, structure\n"
+                                      " - url, format, structure, compression_method\n"
+                                      " - url, access_key_id, secret_access_key\n"
+                                      " - url, access_key_id, secret_access_key, session_token\n"
+                                      " - url, access_key_id, secret_access_key, format\n"
+                                      " - url, access_key_id, secret_access_key, session_token, format\n"
+                                      " - url, access_key_id, secret_access_key, format, structure\n"
+                                      " - url, access_key_id, secret_access_key, session_token, format, structure\n"
+                                      " - url, access_key_id, secret_access_key, format, structure, compression_method\n"
+                                      " - url, access_key_id, secret_access_key, session_token, format, structure, compression_method\n"
+                                      "All signatures supports optional headers (specified as `headers('name'='value', 'name2'='value2')`)";
+};
+
+struct GCSDefinition
+{
+    static constexpr auto name = "gcs";
+    static constexpr auto storage_type_name = "GCS";
+    static constexpr auto signature = S3Definition::signature;
+};
+
+struct COSNDefinition
+{
+    static constexpr auto name = "cosn";
+    static constexpr auto storage_type_name = "COSN";
+    static constexpr auto signature = S3Definition::signature;
+};
+
+struct OSSDefinition
+{
+    static constexpr auto name = "oss";
+    static constexpr auto storage_type_name = "OSS";
+    static constexpr auto signature = S3Definition::signature;
+};
+
+struct HDFSDefinition
+{
+    static constexpr auto name = "hdfs";
+    static constexpr auto storage_type_name = "HDFS";
+    static constexpr auto signature = " - uri\n"
+                                      " - uri, format\n"
+                                      " - uri, format, structure\n"
+                                      " - uri, format, structure, compression_method\n";
+};
+
+template <typename Definition, typename Configuration>
+class TableFunctionObjectStorage : public ITableFunction
+{
+public:
+    static constexpr auto name = Definition::name;
+    static constexpr auto signature = Definition::signature;
+
+    static size_t getMaxNumberOfArguments() { return 8; }
```

Comment:
I think we should add `static constexpr auto max_number_of_arguments` in each defenition and use it here